public/app/features/dashboard-scene/scene/layout-default/DefaultGridLayoutManager.tsx

```diff
@@ -0,0 +1,307 @@
+import {
+  SceneObjectState,
+  SceneGridLayout,
+  SceneObjectBase,
+  SceneGridRow,
+  VizPanel,
+  sceneGraph,
+  sceneUtils,
+  SceneComponentProps,
+} from '@grafana/scenes';
+import { GRID_COLUMN_COUNT } from 'app/core/constants';
+
+import {
+  forceRenderChildren,
+  getPanelIdForVizPanel,
+  NEW_PANEL_HEIGHT,
+  NEW_PANEL_WIDTH,
+  getVizPanelKeyForPanelId,
+} from '../../utils/utils';
+import { DashboardGridItem } from '../DashboardGridItem';
+import { RowActions } from '../row-actions/RowActions';
+import { DashboardLayoutManager } from '../types';
+
+interface DefaultGridLayoutManagerState extends SceneObjectState {
+  grid: SceneGridLayout;
+}
+
+/**
+ * State manager for the default grid layout
+ */
+export class DefaultGridLayoutManager
```

Comment:
Not for this PR, but I think we gonna need an interface for other layout managers to implement common behaviors like `addPanel`/`duplicatePanel` etc. Also, not sure if we should fixate on "panel" naming, i.e. in canvas it's gonna probably be "element" or something like this. But again - let's consider this but not add to this PR.

Reply:
> but I think we gonna need an interface for other layout managers to implement common behaviors like addPanel/duplicatePanel etc

@dprokop DefaultGridLayoutManager does implement DashboardLayoutManager interface which does have addPanel, duplicate panel etc and more, so not sure sure what you mean. 

regarding element vs panel, yes, in the my original POC the layout manager manged "layout elements" (CanvasElement for canvas, DashboardGridItem for default grid, etc) and these layout elements could have different containing objects (panels, buttons etc). 

But in an effort to keep this PR small let the DashboardLayoutManager interface deal directly with VizPanels, as that is what so much code everywhere expects. But hope we can add the LayoutElement abstraction in a future PR and the concept of LayoutElements containing different object types. But needs a lot of changes to panel edit, and persistence

Reply:
Yeah- realized that when reviewing and reflected in my comment 🙈